### PR TITLE
Clarify auth scheme

### PR DESCRIPTION
**Pull request by realloc**
And some minor heading level changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ image](https://hub.docker.com/r/nspccdev/neofs-http-gw) provided for released
 (and occasionaly unreleased) versions of gateway (`:latest` points to the
 latest stable release).
 
-### Notable make targets
+## Notable make targets
```
Comment (roman-khimov) [inline README.md:31]: This was at `###` intentionally before Docker paragraph had been added. Maybe we can swap Docker/make paragraphs.

File: README.md [hunk 2]
```diff
@@ -54,13 +54,13 @@ $ neofs-http-gw -p 192.168.130.72:8080
 $ HTTP_GW_PEERS_0_ADDRESS=192.168.130.72:8080 neofs-http-gw
 ```
 
-### Configuration
+## Configuration
```
Comment (roman-khimov) [inline README.md:57]:
```
## Installation
## Execution
## HTTP API provided
```
I'm not sure it's worth moving this one level up.

File: README.md [hunk 3]
```diff
@@ -285,13 +284,28 @@ object ID, like this:
...
+objects into), but for restricted containers you need to explicitly allow PUT
+operations for request signed with your HTTP Protocol Gateway keys.
+
+If your don't want to manage gateway's secret keys and adjust eACL rules when
+gateway configuration changes (new gate, key rotation, etc) or you plan to use
+public services, there is an option to let your application backend (or you) to
+issue Bearer Tokens ans pass them from the client via gate down to NeoFS level
+to grant access.
+
+NeoFS Bearer Token basically is a container owner-signed ACL data(refer to NeoFS
```
Comment (roman-khimov) [inline README.md:296]: Missing space before `(`.

File: README.md [hunk 4]
```diff
@@ -178,9 +178,8 @@ supported.
 Basic downloading involves container and object ID and is done via GET
 requests to `/get/$CID/$OID` path, like this:
 
-```
+```sh
```
Comment (roman-khimov) [inline README.md:181]: That's something different.

File: README.md (hunk 2 above)
Comment (realloc) [reply README.md:57]: I'm sure it makes the README more readable, especially when you try to navigate with ToC button.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**